### ISSUE TITLE
jOOQ-codegen xml bind fails with 3.11.12 + jdk 11

### ISSUE THREAD

**Issue by almeidaah** [jOOQ/jOOQ#9066]
### Expected behavior and actual behavior:

Expected behavior : 
Generate successfully jOOQ model classes based on database configuration.

Actual behavior:
When running Gradle task 'generateJOOQMapping', I'm receiving this error:

```
org.jooq.codegen.GeneratorException: Error while reading XML configuration
	at org.jooq.codegen.GenerationTool.load(GenerationTool.java:931) ~[jooq-codegen-3.11.12.jar:?]
	at org.jooq.codegen.GenerationTool.main(GenerationTool.java:194) [jooq-codegen-3.11.12.jar:?]
Caused by: javax.xml.bind.JAXBException
	at javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:241) ~[jaxb-api-2.2.12.jar:?]
	at javax.xml.bind.ContextFinder.find(ContextFinder.java:477) ~[jaxb-api-2.2.12.jar:?]
	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:656) ~[jaxb-api-2.2.12.jar:?]
	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:599) ~[jaxb-api-2.2.12.jar:?]
	at org.jooq.codegen.GenerationTool.load(GenerationTool.java:914) ~[jooq-codegen-3.11.12.jar:?]
	... 1 more
Caused by: java.lang.ClassNotFoundException: com.sun.xml.internal.bind.v2.ContextFactory
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583) ~[?:?]
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[?:?]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?]
	at javax.xml.bind.ContextFinder.safeLoadClass(ContextFinder.java:594) ~[jaxb-api-2.2.12.jar:?]
	at javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:239) ~[jaxb-api-2.2.12.jar:?]
	at javax.xml.bind.ContextFinder.find(ContextFinder.java:477) ~[jaxb-api-2.2.12.jar:?]
	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:656) ~[jaxb-api-2.2.12.jar:?]
	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:599) ~[jaxb-api-2.2.12.jar:?]
	at org.jooq.codegen.GenerationTool.load(GenerationTool.java:914) ~[jooq-codegen-3.11.12.jar:?]
	... 1 more
```

### Steps to reproduce the problem 

1 - Using a gradle task 'generateJOOQMapping':

```
task generateJOOQMapping(type:JavaExec, dependsOn:generateDB){
    classpath = sourceSets.main.runtimeClasspath
    main = 'org.jooq.codegen.GenerationTool'
    args 'src/dev/db_utils/jooq-mapping-generation.xml'
}
```

### Versions:

- jOOQ:v 3.11.12
- Java: JDK 11/12
- Database (include vendor): Derby Database
- OS: MacOS
- JDBC Driver (include name if inofficial driver): org.jooq.meta.derby.DerbyDatabase

### PS:
Before having this XML bind exception, the @Generated annotation was not being recognized by java 9+, so I've added to my build.gradle the following dependency:

```
        compile group: 'javax.annotation', name: 'jsr250-api', version: '1.0' 
```

Seems related to https://github.com/jOOQ/jOOQ/issues/7734

My `jooq-mapping.generation.xml` file is here: https://pastebin.com/TECmnfzL/?e=1

**Comment by lukaseder** [comment]
Thank you very much for your report. Starting with JDK 11, JAXB is no longer shipped with the JDK, so if libraries like jOOQ have a JAXB dependency, you will need to supply an implementation yourself. We'll fix this in jOOQ 3.12, which will remove the JAXB implementation dependency entirely as we rolled our own implementation for the few purposes we needed it for.

In Maven, you would typically use a dependency configuraiton like this one:

```xml
                            
                    <dependency>
                        <groupId>com.sun.xml.bind</groupId>
                        <artifactId>jaxb-core</artifactId>
                        <version>${jaxb.version}</version>
                    </dependency>
                    <dependency>
                        <groupId>com.sun.xml.bind</groupId>
                        <artifactId>jaxb-impl</artifactId>
                        <version>${jaxb.version}</version>
                    </dependency>
                    <dependency>
                        <groupId>com.sun.activation</groupId>
                        <artifactId>javax.activation</artifactId>
                        <version>${javax.activation.version}</version>
                    </dependency>
```

I hope this helps

**Comment by almeidaah** [comment]
Thank you Lukas. Since I'm using Gradle, I'll insert these dependencies for now.
